File: blog-admin/blog_admin/blog_admin/models/experience.py
from django.db import models


class Experience(models.Model):
    company = models.CharField(max_length=255)
    position = models.CharField(max_length=255)
    description = models.TextField()
    start_date = models.DateField()
    end_date = models.DateField(blank=True, null=True)
    location = models.CharField(max_length=255, blank=True, null=True)
    company_url = models.CharField(max_length=500, blank=True, null=True)
    sort_order = models.IntegerField(default=0)
    is_active = models.BooleanField(default=True)
    created_at = models.DateTimeField(auto_now_add=True)

    class Meta:
        managed = False
        db_table = 'experiences'
        ordering = ['sort_order', '-start_date']

    def __str__(self):
        end = self.end_date or "Present"
        return f"{self.position} at {self.company} ({self.start_date} – {end})"

File: blog-api/src/test/scala/su/wps/blog/repositories/models/Experience.scala
package su.wps.blog.repositories.models

import cats.syntax.functor.*
import doobie.*
import doobie.generic.auto.*
import su.wps.blog.instances.time.*
import su.wps.blog.tools.types.*

import java.time.{LocalDate, ZonedDateTime}

final case class Experience(
  id: PosInt,
  company: Varchar[W.`255`.T],
  position: Varchar[W.`255`.T],
  description: Varchar[W.`1000`.T],
  startDate: LocalDate,
  endDate: Option[LocalDate],
  location: Option[Varchar[W.`255`.T]],
  companyUrl: Option[Varchar[W.`500`.T]],
  sortOrder: Int,
  isActive: Boolean,
  createdAt: ZonedDateTime
)

object Experience {

  object sql {
    def insert(experience: Experience): ConnectionIO[Unit] = {
      val sql =
        """
          |INSERT INTO experiences (
          | id,
          | company,
          | position,
          | description,
          | start_date,
          | end_date,
          | location,
          | company_url,
          | sort_order,
          | is_active,
          | created_at
          |) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
          |""".stripMargin
      Update[Experience](sql).run(experience).void
    }
  }
}
